Getting a panic when running this locally:

```
panic: assignment to entry in nil map

goroutine 231 [running]:
github.com/grafana/grafana/pkg/services/ngalert/state.(*cache).set(0xc000a8cac0, 0xc0010108f0)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/services/ngalert/state/cache.go:87 +0x12c
github.com/grafana/grafana/pkg/services/ngalert/state.(*Manager).set(...)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/services/ngalert/state/manager.go:41
github.com/grafana/grafana/pkg/services/ngalert/state.(*Manager).Put(0xc0008c62d0, 0xc000db1400, 0x26, 0x40)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/services/ngalert/state/manager.go:127 +0x47
github.com/grafana/grafana/pkg/services/ngalert/schedule.(*schedule).WarmStateCache(0xc000a850e0, 0xc0008c62d0)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/services/ngalert/schedule/schedule.go:404 +0xbc7
github.com/grafana/grafana/pkg/services/ngalert.(*AlertNG).Run(0xc000122300, 0x30ce5e8, 0xc000638540, 0x21f22c07be4a3a0a, 0xdae0fb596a0c8677)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/services/ngalert/ngalert.go:100 +0x7e
github.com/grafana/grafana/pkg/server.(*Server).Run.func1(0xf02892008dd7e253, 0x4b15382ac9fbf4f5)
	/home/kbrandt/go/github.com/grafana/grafana/pkg/server/server.go:203 +0xa6
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000aa6ea0, 0xc0003db4d0)
	/home/kbrandt/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/kbrandt/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

--- a/pkg/services/ngalert/state/cache.go
+++ b/pkg/services/ngalert/state/cache.go
@@ -15,7 +15,7 @@ import (
 )
 
 type cache struct {
-	states    map[int64]map[string]map[string]*State
+	states    map[int64]map[string]map[string]*State // orgID > alertRuleUID > stateID > state
 	mtxStates sync.RWMutex
 	log       log.Logger
 	metrics   *metrics.Metrics

--- a/pkg/services/ngalert/state/cache.go
+++ b/pkg/services/ngalert/state/cache.go
@@ -82,6 +82,8 @@ func (c *cache) set(entry *State) {
 	defer c.mtxStates.Unlock()
 	if _, ok := c.states[entry.OrgID]; !ok {
 		c.states[entry.OrgID] = make(map[string]map[string]*State)
+	}
+	if _, ok := c.states[entry.OrgID][entry.AlertRuleUID]; !ok {
 		c.states[entry.OrgID][entry.AlertRuleUID] = make(map[string]*State)
 	}
 	c.states[entry.OrgID][entry.AlertRuleUID][entry.CacheId] = entry

